packages/eyeglass/src/assets/Assets.ts

```diff
@@ -1,242 +1,275 @@
+interface Installs {
+  install: Installer;
+}
+
+export default class Assets implements Resolves, Installs {
+  sassUtils: any;
+  eyeglass: IEyeglass;
+  /**
+   * Assets declared by the application.
+   */
+  collection: AssetsCollection;
+  /**
+   * Assets declared by eyeglass modules.
+   */
+  moduleCollections: Array<AssetsCollection>;
+  AssetCollection: () => AssetsCollection;
+  AssetPathEntry: (src: string, options: AssetSourceOptions) => AssetsSource;
+  constructor(eyeglass: IEyeglass, sass: SassImplementation) {
+    this.sassUtils = require("node-sass-utils")(sass);
+    this.eyeglass = eyeglass;
+    // create a master collection
+    this.collection = new AssetsCollection(eyeglass.options);
+    // and keep a list of module collections
+    this.moduleCollections = [];
+
+    // Expose these temporarily for back-compat reasons
+    function deprecate(method: string) {
+      eyeglass.deprecate("0.8.3", "0.9.0", [
+        "The assets." + method + " interface will be removed from the public API.",
+        "If you currently use this method, please open an issue at",
+        "https://github.com/sass-eyeglass/eyeglass/issues/ so we can",
+        "understand and accommodate your use case"
+      ].join(" "));
+    }
+    this.AssetCollection = function () {
+      deprecate("AssetCollection");
+      return new AssetsCollection(eyeglass.options);
+    };
+    this.AssetPathEntry = function (src: string, options: AssetSourceOptions) {
+      deprecate("AssetPathEntry");
+      return new AssetsSource(src, options);
+    };
+  }
+
+  /**
+    * @see AssetsCollection#asAssetImport
+    */
+  asAssetImport(name: string | undefined): string {
+    return this.collection.asAssetImport(name);
+  }
+
+  /**
+    * @see AssetsCollection#addSource
+    */
+  addSource(src: string, opts: AssetSourceOptions): AssetsCollection {
+    return this.collection.addSource(src, opts);
+  }
+
+  /**
+    * @see AssetsCollection#cacheKey
+    */
+  cacheKey(name: string): string {
+    return this.collection.cacheKey(name);
+  }
+  /**
+    * creates a new AssetsCollection and adds the given source
+    * @see #addSource
+    * @param    {String} src - the source directory
+    * @param    {Object} opts - the options
+    * @returns  {AssetsCollection} the instance of the AssetsCollection
+    */
+  export(src: string, opts: AssetSourceOptions): AssetsCollection {
+    let assets = new AssetsCollection(this.eyeglass.options);
+    this.moduleCollections.push(assets);
+    return assets.addSource(src, opts);
+  }
+  /**
+    * resolves an asset given a uri
+    * @param    {SassMap} $assetsMap - the map of registered Sass assets
+    * @param    {SassString} $uri - the uri of the asset
+    * @param    {Function} cb - the callback that is invoked when the asset resolves
+    */
+  resolveAsset($assetsMap: SassMap, $uri: SassValue, cb: (error: Error | null, uri?: string, file?: string) => any) {
```

Comment:
This particular API is documenting that it only returns `Error | null` to the callback in the case of an error. Where I can control what is returned, I think `Error | null` is the better callback signature vs `any` (actually I think `unknown` is a better type than `any` in this case because `any` disables type checking where `unknown` treats it as a union of all possible types.)

That said, I am making this mistake on some of the public facing APIs (E.g. https://github.com/sass-eyeglass/eyeglass/blob/type_annotations/packages/eyeglass/src/assets/Assets.ts#L22 & https://github.com/sass-eyeglass/eyeglass/blob/type_annotations/packages/eyeglass/src/assets/Assets.ts#L30) So I will review all the public facing callback signatures and update the type definitions appropriately.

Reply:
Fixed in 921492d1b64a64499a49ce31665e3153ecb9dabc.